File: Algorithms/121. Best Time to Buy and Sell Stock.c
int maxProfit(int* prices, int pricesSize) {
    int res = 0,p=0;

    if (!prices) {
        return res;
    }

    int f1 = prices[0];

    for (size_t i = 0; i < pricesSize; i++) {
        if (prices[i] < f1) {
            f1 = prices[i]
        }else if (prices[i] > f1 && ((p=prices[i] - f1) > res )) {
            res = p;
        }
    }

    return res;
}

File: Algorithms/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        res = 0

        if len(prices) == 0:
            return res

        f1 = prices[0]

        for i in prices:
            if i < f1:
                f1 = i
            elif (i > f1) and (i-f1 > res):
                res = i-f1

        return res
